scripts/profiler/rtt2socket.py

```diff
@@ -0,0 +1,236 @@
+    def disconnect(self):
+        self.stop_logging_events()
+        # read remaining data to buffer
+        bufs = bytearray()
+        while True:
+            try:
+                buf = self.jlink.rtt_read(self.rtt_up_channels['data'],
+                                          self.config['rtt_read_chunk_size'],
+                                          encoding=None)
+
+            except APIError:
+                self.logger.error("Problem with reading RTT data")
+                buf = []
+
+            if len(buf) > 0:
+                bufs += buf
+            else:
+                if len(bufs) > 0:
+                    try:
+                        self.sock_ev.sendto(bufs, SOCK_RECV_EV)
+                    except socket.error:
+                        self.logger.info("Unable to send remaining events")
+                break
+
+        try:
+            self.jlink.rtt_stop()
+            self.jlink.disconnect_from_emu()
+            self.jlink.close()
+
+        except APIError:
+            self.logger.error("JLink connection lost")
+            return
+
+        self.logger.info("Disconnected from device")
+
+    def read_bytes(self):
+        bufs = bytearray()
+        while True:
+            try:
+                buf = self.jlink.rtt_read(self.rtt_up_channels['data'],
+                                          self.config['rtt_read_chunk_size'],
+                                          encoding=None)
+            except APIError:
+                self.logger.error("Problem with reading RTT data")
+                self.disconnect()
+                sys.exit()
+
+            if len(buf) > self.config['rtt_additional_read_thresh']:
+                bufs += buf
+                continue
+
+            if len(buf) > 0:
+                bufs += buf
+                break
+
+            if self.finish_event is not None and self.finish_event.is_set():
+                self.finish_event.clear()
+                self.logger.info("Real time transmission closed")
+                self.disconnect()
+                sys.exit()
+
+            time.sleep(0.05)
+
+        return bufs
+
+    def _read_all_events_descriptions(self):
+        self._send_command(Command.INFO)
+        desc_buf = ""
+        # Empty field is send after last event description
+        while '\n\n' not in desc_buf:
+            try:
+                buf_temp = self.jlink.rtt_read(self.rtt_up_channels['info'],
+                                    self.config['rtt_read_chunk_size'],
+                                    encoding='utf-8')
+
+            except APIError:
+                self.logger.error("Problem with reading RTT data")
+                self.disconnect()
+                sys.exit()
+
+            desc_buf += buf_temp
+            time.sleep(0.1)
+        return desc_buf
+
+    def read_and_transmit_data(self):
+        desc_buf = self._read_all_events_descriptions()
+        try:
+            self.sock_desc.sendto(desc_buf.encode(), SOCK_RECV_DESC)
```

Comment:
we could work with bytes (or bytearray for performance reasons) all the time so no encoding would be needed